nixos/modules/services/networking/wstunnel.nix

```
@@ -0,0 +1,341 @@
+        type = types.nullOr types.path;
+        default = null;
+        example = "/var/lib/acme/example.com/key.pem";
+      };
+
+      verboseLogging = mkOption {
+        description = mdDoc "Enable verbose logging.";
+        type = types.bool;
+        default = false;
+      };
+
+      runAsRoot = mkOption {
+        description = mdDoc "Run `wstunnel` as root.";
+        type = types.bool;
+        default = false;
+      };
+
+      package = mkOption {
+        description = mdDoc "The `wstunnel` package to use for this instance.";
+        type = types.package;
+        default = cfg.package;
+        defaultText = literalExpression "services.wstunnel.package";
+      };
+
+      extraArgs = mkOption {
+        description = mdDoc "Extra command line arguments to pass to `wstunnel`.";
+        type = types.str;
+        default = "";
+      };
+    };
+  };
+  clientOptions = {
+    options = {
+      address = mkOption {
+        description = mdDoc "Server address to connect to.";
+        type = types.str;
+        example = "example.com";
+      };
+
+      port = mkOption {
+        description = mdDoc "Server port to connect to.";
+        type = types.port;
+      };
+
+      enableHTTPS = mkOption {
+        description = mdDoc "Enable HTTPS when connecting to the server.";
+        type = types.bool;
+        default = false;
+      };
+
+      autoStart = mkOption {
+        description = mdDoc "Whether this tunnel server should be started automatically.";
+        type = types.bool;
+        default = true;
+      };
+
+      localToRemote = mkOption {
+        description = mdDoc "Hosts and ports to listen on, plus the remote hosts and ports to forward traffic to.";
+        type = types.listOf types.str;
+        default = [];
+        example = [ "127.0.0.1:8080:127.0.0.1:8080" ];
+      };
+
+      dynamicToRemote = mkOption {
+        description = mdDoc "Host and port for the SOCKS5 proxy to dynamically forward traffic to. Leave this at `null` to disable the SOCKS5 proxy.";
+        type = types.nullOr types.str;
+        default = null;
+        example = [ "127.0.0.1:1080" ];
+      };
+
+      udp = mkOption {
+        description = mdDoc "Whether to forward UDP instead of TCP traffic.";
+        type = types.bool;
+        default = false;
+      };
+
+      udpTimeout = mkOption {
+        description = mdDoc "When using UDP forwarding, timeout in seconds after which the tunnel connection is closed. `-1` means no timeout.";
+        type = types.int;
+        default = 30;
+      };
+
+      httpProxy = mkOption {
+        description = mdDoc "Proxy to use to connect to the wstunnel server (`USER:PASS@HOST:PORT`). WARNING: Passwords specified here will be world-readable in the Nix store! To safely pass a password to the service, point the `environmentFile` option to a file containing `PROXY_PASSWORD=<your-password-here>` and set this option to `<user>:$PROXY_PASSWORD@<host>:<port>`.";
```

```suggestion
        description = mdDoc ''
          Proxy to use to connect to the wstunnel server (`USER:PASS@HOST:PORT`).
          
          WARNING: Passwords specified here will be world-readable in the Nix store! To safely pass a password to the service, point the `environmentFile` option to a file containing `PROXY_PASSWORD=<your-password-here>` and set this option to `<user>:$PROXY_PASSWORD@<host>:<port>`.
        '';
```
please lookup the doc warning syntax to have a yellow box